src/ethereum/api/Bridge.ts

```diff
@@ -22,6 +24,28 @@ export class Bridge {
       collections.push(collection)
     }
 
+    const remoteCollectionCreators = remoteCollections.map((collection) =>
+      getAddress(collection.creator)
+    )
+    const dbCollectionsByRemotes = await Collection.findByEthAddresses(
+      remoteCollectionCreators
+    )
+    for (const remoteCollection of remoteCollections) {
+      // Get only db collections that has not been added in the collection array
+      const dbCollection = dbCollectionsByRemotes.find(
+        (collection) =>
+          collection.contract_address === remoteCollection.id &&
+          !collections.some(
+            (collection) => collection.contract_address === remoteCollection.id
+          )
```

Comment:
This seems...wrong. I don't really get what this condition is doing, but the first thing that seems odd is that you're doing a some (iterating) for each `dbCollectionsByRemotes`, but the result is always the same, as the collection variable is shadowing the one on the iteration (some)

At the very least, maybe we should move  this outside the find?

Reply:
Was not in my best dev momento. What a shame